### PR TITLE
[WIP] [DOCS] Update non-existent pragma 0.4.99 to 0.5.0

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Closes https://github.com/ethereum/solidity/issues/3379 and specifically https://github.com/ethereum/solidity/issues/3379#issuecomment-450906353 to ensure that all code examples use up to dat pragma versions.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/050-breaking-changes.rst [hunk 1]
```diff
@@ -308,7 +308,7 @@ This will no longer compile with Solidity v0.5.0. However, you can define a comp
 
 ::
 
-   pragma solidity >0.4.99 <0.6.0;
+   pragma solidity >0.5.0 <0.6.0;
```
Comment (axic) [inline docs/050-breaking-changes.rst]: Should be `>=0.5.0 <0.6.0` or just `^0.5.0`

Comment (axic) [reply docs/050-breaking-changes.rst]: Though the previous `>0.4.99 <0.6.0` is actually valid.

Comment (ChrisChinchilla) [reply docs/050-breaking-changes.rst]: @axic I had barely pushed the branch ;)

Comment (ChrisChinchilla) [reply docs/050-breaking-changes.rst]: @axic It is? Then I am slightly confused now, was based on this comment - https://github.com/ethereum/solidity/issues/3379#issuecomment-448675334

Comment (axic) [reply docs/050-breaking-changes.rst]: Had a look at the referenced issue. I think for these cases the proper solution is to use `pragma solidity ^0.5.0`

Comment (ChrisChinchilla) [reply docs/050-breaking-changes.rst]: OK @axic updated again :)

File: docs/050-breaking-changes.rst [hunk 2]
```diff
@@ -308,7 +308,7 @@ This will no longer compile with Solidity v0.5.0. However, you can define a comp
 
 ::
 
-   pragma solidity >0.4.99 <0.6.0;
+   pragma solidity ^0.5.0 <0.6.0;
```
Comment (chriseth) [inline docs/050-breaking-changes.rst]: I guess `<0.6.0` could be removed.

**Comment by ChrisChinchilla** [comment]
@chriseth Updated, rebased, squashed

**Review by chriseth — APPROVED** [review]

**Pull request merged**